cli/CommandExecutor.hpp

```diff
@@ -0,0 +1,97 @@
+/**
+ *   Copyright 2016, Quickstep Research Group, Computer Sciences Department,
+ *     University of Wisconsin—Madison.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ **/
+
+#ifndef QUICKSTEP_CLI_COMMAND_COMMAND_EXECUTOR_HPP_
+#define QUICKSTEP_CLI_COMMAND_COMMAND_EXECUTOR_HPP_
+
+#include <cstdio>
+#include <string>
+
+#include "parser/ParseStatement.hpp"
+#include "utility/Macros.hpp"
+
+using std::fprintf;
+using std::fputc;
+using std::size_t;
+using std::string;
+
+namespace quickstep {
+
+class CatalogDatabase;
+class CatalogAttribute;
+class CatalogRelation;
+
+
+/** \addtogroup CLI
+ *  @{
+ */
+
+/**
+ * @brief Execution handlers for various quickstep cli commands
+ */
+class CommandExecutor {
+ public:
+  /**
+   * @brief Executes the command by calling the command handler.
+   *        
+   * @param statement The parsed statement from the cli.
+   * @param catalog_database The catalog information about the current database.
+   * @param out The stream where the output of the command has to be redirected to.      
+   */
+  static void executeCommand(const ParseStatement &statement,
+                             const CatalogDatabase &catalog_database,
+                             FILE *out);
+
+ private:
+  static const std::size_t kInitMaxColumnWidth;
+  CommandExecutor();
```

Comment:
Prefer to use global functions, instead of the current implementation:

https://google.github.io/styleguide/cppguide.html#Nonmember,_Static_Member,_and_Global_Functions

With that being said, I suggest the following changes:
```C++
// .hpp file
#include <cstddef>  // for std::size_t

namespace quickstep {
namespace cli {

// Comments for why choose the value.
constexpr std::size_t kInitMaxColumnWidth = 6;

static const char *kDescribeDatabaseCommand;
static const char *kDescribeTableCommand;

/*
 * @brief ...
 */
void ExecuteCommand(...);

}  // namespace cli
}  // namespace quickstep
```

```C++
// .cpp file
namespace quickstep {
namespace cli {

const char *kDescribeDatabaseCommand = "\\d";
const char *kDescribeTableCommand = "\\dt";

namespace {

void ExecuteDescribeTable(...) {
  ...
}

void ExecuteDescribeTable() {
  ...
}

}  // namespace

void ExecuteCommand(...) {
  ...
}

}  // namespace cli
}  // namespace quickstep
```